### ISSUE TITLE
Cannot Reproduce Model on CIFAR with non-OOD sample generated

### ISSUE THREAD

**Issue by zxy19** [deeplearning-wisc/dream-ood#25]
Hello, I met some problem when reproduce the model on CIFAR dataset. When I run the scripts as instructions of the readme, some non-OOD samples are generated by `dream_ood.py` and the model trained by these samples will have an very high error rate and bad metrics.
It seem that the weights on ImageNet100 is OK while the weights on cifar100 gives an incorrect result.

---

**Problem:** Unable to reproduce the model on CIFAR dataset  
**Expected:** All generated samples should be OOD samples, and after OOD training, the model should normally distinguish OOD samples and achieve metrics close to those in the paper  
**Actual Result:** Some generated samples do not belong to OOD samples, and after training, a very high ErrorRate (greater then 90%) is obtained

### Self Pre-training
1. Execute `pretrain_cifar100.py` to obtain `id_feat_cifar100_199epoch.npy`
2. Execute `get_embed_cifar100.py` to obtain `cifar100_outlier_npos_embed_noise_0.07_select_50_KNN_300.npy`
3. Execute `dream_ood.py --plms --n_iter 50 --n_samples 3 --outdir ./output/dreamood/ --loaded_embedding ./cifar100_outlier_npos_embed_noise_0.07_select_50_KNN_300.npy --ckpt ./pretrained/sd-v1-4.ckpt --id_data cifar100 --skip_grid` to get output
At this point, it can be noticed that some generated samples are not OOD samples

<img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/0d69c850-429c-4ddb-9ec4-49d825a8a489" />
31/elephant_00016.png for example.

### Using Pretrained `Outliers for Cifar-100` File
0. Download from section Training.1: https://drive.google.com/file/d/18s3yozpejwYm7tx89JONbSSV-c4z4vYT/view?usp=sharing
1. Execute `pretrain_cifar100.py` with parameter `--epochs=0`, and remove the scheduler from the code, obtain `id_feat_cifar100_199epoch.npy`
Then do the steps same as above

### OOD Training
Using the above incorrect data for OOD training, results can be obtained as following:

```log
[np.float64(0.09999999983943042)]
loss energy is:  0.3527112483978272
Epoch   1 | Time     5 | Train Loss 1.3166 | Test Loss 45.134 | Test Error 99.00
[np.float64(0.09999999935772169)]
loss energy is:  0.7307663917541505
Epoch   2 | Time     2 | Train Loss 1.7674 | Test Loss 208.093 | Test Error 99.00
[np.float64(0.09999999855487378)]
loss energy is:  1.8636436462402344
Epoch   3 | Time     2 | Train Loss 3.0108 | Test Loss 4338.625 | Test Error 99.00
[np.float64(0.09999999743088671)]
loss energy is:  0.8160270690917969
Epoch   4 | Time     2 | Train Loss 1.9134 | Test Loss 8212.243 | Test Error 99.00
[np.float64(0.09999999598576051)]
loss energy is:  0.8066923141479493
Epoch   5 | Time     2 | Train Loss 2.0616 | Test Loss 19825.307 | Test Error 99.00
[np.float64(0.09999999421949517)]
loss energy is:  0.531214189529419
Epoch   6 | Time     2 | Train Loss 1.6718 | Test Loss 63568.966 | Test Error 99.00
[np.float64(0.0999999921320907)]
loss energy is:  0.41709928512573247
Epoch   7 | Time     2 | Train Loss 1.4468 | Test Loss 337140.581 | Test Error 99.00
[np.float64(0.0999999897235471)]
loss energy is:  0.33693962097167973
Epoch   8 | Time     2 | Train Loss 1.5084 | Test Loss 313184.054 | Test Error 99.00
[np.float64(0.09999998699386442)]
loss energy is:  0.47394585609436035
Epoch   9 | Time     2 | Train Loss 1.7264 | Test Loss 1744827531264.000 | Test Error 99.00
[np.float64(0.09999998394304266)]
loss energy is:  0.5276129245758057
Epoch  10 | Time     2 | Train Loss 1.8515 | Test Loss 1306506384.000 | Test Error 99.00
[np.float64(0.09999998057108184)]
loss energy is:  0.48396916389465333
Epoch  11 | Time     2 | Train Loss 2.1886 | Test Loss 154805239.400 | Test Error 99.00
[np.float64(0.09999997687798197)]
loss energy is:  0.5735450267791748
Epoch  12 | Time     2 | Train Loss 1.8302 | Test Loss 3192300.019 | Test Error 99.00
[np.float64(0.09999997286374307)]
loss energy is:  0.3393503427505493
Epoch  13 | Time     2 | Train Loss 1.5255 | Test Loss 1848225.337 | Test Error 99.00
[np.float64(0.09999996852836521)]
loss energy is:  0.32673354148864747
Epoch  14 | Time     2 | Train Loss 1.3364 | Test Loss 742353.412 | Test Error 99.00
[np.float64(0.0999999638718484)]
loss energy is:  0.3266267776489258
Epoch  15 | Time     2 | Train Loss 1.4531 | Test Loss 272735.729 | Test Error 99.00
[np.float64(0.09999995889419261)]
loss energy is:  0.3275835752487183
Epoch  16 | Time     2 | Train Loss 1.2994 | Test Loss 84012.948 | Test Error 99.00
[np.float64(0.09999995359539796)]
loss energy is:  0.3254929780960083
Epoch  17 | Time     2 | Train Loss 1.3039 | Test Loss 37641.417 | Test Error 99.00
[np.float64(0.09999994797546444)]
loss energy is:  0.3308654069900513
Epoch  18 | Time     2 | Train Loss 1.3032 | Test Loss 11186.735 | Test Error 99.00
[np.float64(0.09999994203439207)]
loss energy is:  0.3389475345611572
Epoch  19 | Time     2 | Train Loss 1.2748 | Test Loss 10327.079 | Test Error 98.99
[np.float64(0.09999993577218091)]
loss energy is:  0.3646801710128784
Epoch  20 | Time     2 | Train Loss 1.2942 | Test Loss 6800.261 | Test Error 99.00
[np.float64(0.09999992918883102)]
loss energy is:  0.4106364727020264
Epoch  21 | Time     2 | Train Loss 1.3440 | Test Loss 3950.483 | Test Error 99.00
[np.float64(0.0999999222843424)]
loss energy is:  0.5685149192810058
Epoch  22 | Time     2 | Train Loss 1.4949 | Test Loss 77.186 | Test Error 99.02
[np.float64(0.09999991505871512)]
loss energy is:  0.36328508853912356
Epoch  23 | Time     2 | Train Loss 1.4871 | Test Loss 27.685 | Test Error 98.94
[np.float64(0.09999990751194925)]
loss energy is:  0.3500572443008423
Epoch  24 | Time     2 | Train Loss 2.3563 | Test Loss 14.540 | Test Error 98.98
[np.float64(0.09999989964404477)]
loss energy is:  0.33606762886047364
```

**Comment by zxy19** [comment]
Sorry that I've misunderstood some parameters in the document, the problem has solved now.
